client/client.go

```diff
@@ -78,10 +83,23 @@ import (
 type ClientConfig struct {
 	// UseDTLS controls whether DTLS should be used to secure the connection
 	// to the MQTT-SN gateway.
-	UseDTLS        bool
-	Certificate    *tls.Certificate
-	PrivateKey     crypto.PrivateKey
-	CACertificates []*x509.Certificate
+	UseDTLS     bool
+	Certificate *tls.Certificate
+	PrivateKey  crypto.PrivateKey
+	// UsePSK controls whether pre-shared key should be used to secure the
+	// connection to the MQTT-SN gateway. If UsePSK is true, you must provide
+	// PSKIdentity, PSKApiBasicAuthUsername, PSKApiBasicAuthPassword and
+	// PSKApiEndpoint.
+	// If UsePSK is true, the client will use PSK instead of the certificate
+	// and private key.
+	UsePSK                  bool
+	PSK                     *cache.Cache
+	PSKCacheExpiration      time.Duration
+	PSKIdentity             string
```

Comment:
Please use `PSKIdentityHint` in config as well to be consistent with the `dtls` library.